ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp

```diff
@@ -0,0 +1,179 @@
+                                          int classes,
+                                          int outputs,
+                                          int batch,
+                                          int location,
+                                          int entry)
+            {
+                int n = location / (width * height);
+                int loc = location % (width * height);
+                return batch * outputs + n * width * height * (coords + classes + 1) +
+                       entry * width * height + loc;
+            }
+
+            template <typename T>
+            static inline T sigmoid(float x)
+            {
+                return static_cast<T>(1.f / (1.f + std::exp(-x)));
+            }
+            template <typename T>
+            static inline void softmax_generic(
+                const T* src_data, T* dst_data, int batches, int channels, int height, int width)
+            {
+                const int area = height * width;
+                for (unsigned int batch_idx = 0; batch_idx < batches; batch_idx++)
+                {
+                    const int offset = batch_idx * channels * area;
+                    for (unsigned int i = 0; i < height * width; i++)
+                    {
+                        T max = src_data[batch_idx * channels * area + i];
+                        for (unsigned int channel_idx = 0; channel_idx < channels; channel_idx++)
+                        {
+                            T val = src_data[offset + channel_idx * area + i];
+                            if (val > max)
+                            {
+                                max = val;
+                            }
+                        }
+
+                        T sum = 0;
+                        for (unsigned int channel_idx = 0; channel_idx < channels; channel_idx++)
+                        {
+                            dst_data[offset + channel_idx * area + i] =
+                                std::exp(src_data[offset + channel_idx * area + i] - max);
+                            sum += dst_data[offset + channel_idx * area + i];
+                        }
+
+                        for (unsigned int channel_idx = 0; channel_idx < channels; channel_idx++)
+                        {
+                            dst_data[offset + channel_idx * area + i] /= sum;
+                        }
+                    }
+                }
+            }
+
+            template <typename T>
+            void region_yolo(const T* input,
+                             T* output,
+                             const Shape& input_shape,
+                             const int coords,
+                             const int classes,
+                             const int regions,
+                             const bool do_softmax,
+                             const std::vector<int64_t>& mask,
+                             const int axis,
+                             const int end_axis)
```

Comment:
I see `axis`, `end_axis` attributes in the [RegionYolo ](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/RegionYolo_1.md)spec, but you don't use them in this implementation. 
Do we need them here?

Reply:
You are right. These attributes are used to partially flatten the output tensor shape (into 2D) if do_softmax=1. As output shape is inferred by validate_and_infer_types() we do not need these parameters in reference implementation.